mergify_engine/clients/http.py

```diff
@@ -64,12 +64,34 @@ class HTTPClientSideError(httpx.HTTPStatusError):
     def message(self) -> str:
         # TODO(sileht): do something with errors and documentation_url when present
         # https://developer.github.com/v3/#client-errors
-        return typing.cast(str, self.response.json()["message"])
+        to_cast = (
+            self.response.json()["errors"][0]["message"]
+            if "errors" in self.response.json()
+            else self.response.json()["message"]
+        )
+        return typing.cast(str, to_cast)
 
     @property
     def status_code(self) -> int:
         return self.response.status_code
 
+    def contains(self, msg_to_find: str) -> bool:
+
+        if "errors" in self.response.json():
+            return any(
+                msg_to_find in error["message"]
+                for error in self.response.json()["errors"]
+            )
+
+        return (
+            True
+            if (
+                msg_to_find in self.response.json()["message"]
+                or msg_to_find in self.response.text
+            )
+            else False
+        )
```

Comment:
It looks like I miss guided you, we don't use response.text anymore, so maybe it's safer to not handle this case

```suggestion
        return msg_to_find in self.response.json()["message"]
```